src/mavlink-router/endpoint.cpp

```diff
@@ -771,22 +796,45 @@ ssize_t UdpEndpoint::_read_msg(uint8_t *buf, size_t len)
 
 int UdpEndpoint::write_msg(const struct buffer *pbuf)
 {
+    if (tx_buf.len + pbuf->len > TX_BUF_MAX_SIZE)
+    {
+        log_debug("Dropping message, tx buffer full");
+        return 0;
+    }
+    memcpy(&tx_buf.data[tx_buf.len], pbuf->data, pbuf->len);
+    tx_buf.len += pbuf->len;
+
```

Comment:
this means you would hypothetically exceed the specified MTU size by up to the amount of a mavlink frame (minus one) -- you need to check for overflow and force flush before appending

structure that I would suggest: two functions:
- force_write(): performs actual write, deschedules timer (if there was any pending) to avoid early send of following data
- schedule_delayed_write(deadline): schedule write to happen latest at deadline of all buffer contents; leaves present timeout if it has earlier deadline, otherwise updates; post-condition: timer scheduled

Reply:
I guess the question is whether it is better to have a 1-message-size conservative threshold for triggering force_write(), and send the packets immediately, or to get the little bit of extra throughput but at the cost of always having that one extra packet being delayed by up to the coalescing time.

I was a little unhappy with the function naming, this is a nicer abstraction of the problem.

Reply:
not sure I understand the first paragraph correctly, but to clarify what I was trying to say: if something is called "buf_max_packet_size", then I would be slightly surprised by behavior that the *real* max packet size is in reality "buf_max_packet_size + X" where X is size of mavlink frame -- it might lead to accidental udp packet fragmentation if someone goes in knowing the "correct" MTU and assumes the code behaves. My comment is about "adhering to what is promised in the name". I don't think you "gain" efficiency but being fuzzy about the API, you just artificially increase the max packet size by an unspecified amount.

Reply:
Ah, yes, my bad. I earlier had an extra variable as `trigger' which was one packet size smaller, but it was removed in a refactoring pass. 

I'll make the behavior reflect the naming.